modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RestRankEvalAction.java

```diff
@@ -191,7 +208,16 @@ public static void parseRankEvalRequest(RankEvalRequest rankEvalRequest, RestReq
             throws IOException {
         List<String> indices = Arrays.asList(Strings.splitStringByCommaToArray(request.param("index")));
         List<String> types = Arrays.asList(Strings.splitStringByCommaToArray(request.param("type")));
-        RankEvalSpec spec = RankEvalSpec.parse(context.parser(), context);
+        RankEvalSpec spec = null;
+        logger.trace("received rank eval request");
+        logger.trace(request.path());
+        if (request.path().contains("template")) {
+            logger.trace("rank eval request is templated");
+            spec = RankEvalSpec.parse(context.parser(), context, true);
```

Comment:
nit: the if-else block could be simplified by using the contains("template") as the method argument. (provided that we don't need the logger here)